core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -113,8 +116,19 @@ private InstanceData getInstanceData() {
       instanceDataFuture = currentInstanceData;
     }
     try {
-      return Uninterruptibles.getUninterruptibly(instanceDataFuture);
-    } catch (ExecutionException ex) {
+      return instanceDataFuture.get(30, TimeUnit.SECONDS);
```

Comment:
This is a small change of behavior: 

The old way hangs application thread until the current request finishes. If the request succeeds, this returns the InstanceData. If not, this throws a RuntimeException. 

The new way waits up to 30 seconds until a refresh attempt succeeds. If a refresh attempt succeeds within the 30 second timeout, the a future containing the latest InstanceData is returned. If not, this throws a RuntimeException using the exception from the last failed refresh attempt.

Reply:
Minor not: could we pull out the 30 into a constant? `MAX_INSTANCE_DATA_WAIT_TIME` or similar?

Reply:
Done.